command-palette.lua: fix scaling behavior

dyphire opened the pull request:
This is essentially a problem in [extended-menu.lua](https://github.com/Seme4eg/mpv-scripts/blob/98cc63757654e7ede2533af2b0921afbc4165dc1/script-modules/extended-menu.lua#L177-L193) script, which uses dynamic PlayResX and PlayResY values, which causes the font size to have to be dynamic as well or the menu won't be rendered correctly.

The `o.font_size = math.floor(initial_font_size * mp.get_property_native("display-hidpi-scale"))` still a one-time set static font size that can't be adapted to different window sizes.

The correct way to use this is to render the menu using fixed PlayResX and PlayResY, such as the relevant code in [playlistmanager.lua](https://github.com/jonniek/mpv-playlistmanager/blob/54ed4421f9db5df22524d37e565fe4bb2d12117e/playlistmanager.lua#L671-L673).
The correct solution is to fix it in extended-menu.lua, but until then use this temporary method.

dyphire commented:
Open a [PR](https://github.com/Seme4eg/mpv-scripts/pull/10) upstream to try to fix it, change the content of this PR after the merge is done.

dyphire commented:
Update: fixed in upstream

stax76 commented:
With this and updated `extended-menu.lua` I get different font sizes depending on window size, with normal size the font size is good, with fullscreen or maximized I get now a font size that is too large. I prefer having the same font size regardless of window size, like it was before.  

Ideally, font size 16 in console.lua should be the same font size on screen as font size 16 in command_palette.lua, it's more user-friendly being consistent.

https://mpv.io/manual/master/#console-font-size

dyphire commented:
> With this and updated `extended-menu.lua` I get different font sizes depending on window size, with normal size the font size is good, with fullscreen or maximized I get now a font size that is too large. I prefer having the same font size regardless of window size, like it was before.

I can't reproduce this, the command panel always takes up half the size of the mpv window, whether in windowed mode or full screen, which seems appropriate to me.
![image](https://github.com/user-attachments/assets/b8e1f640-b374-4dcb-b123-8d64559740c5)
![image](https://github.com/user-attachments/assets/4d3bc4f2-975f-4604-88d7-ba01ee0f1107)
 
> Ideally, font size 16 in console.lua would be the same font size on screen as font size 16 in command_palette.lua, it's more user-friendly being consistent.
> 
> https://mpv.io/manual/master/#console-font-size

The font size of 16 was too small, as was the console, and I changed it to 30 to have a more suitable display.
![image](https://github.com/user-attachments/assets/ac42977d-dc18-45bd-9e3a-ca1eea82b6d0)
![image](https://github.com/user-attachments/assets/5aa99502-93f4-407d-8a60-c6159154fb8c)
![image](https://github.com/user-attachments/assets/c696aec0-dfef-4dcb-a449-810e96a2dc15)

stax76 commented:
Using a 4K display, font size 16 with normal window size is too small, many people have poor eyes, me too.

![normal16](https://github.com/user-attachments/assets/5a4441ff-d361-4f95-a14b-20988495b2eb)

When the window is maximized, the font size is okay, the problem is normal and maximized window have different font sizes, Before it was the same.

![maximized16](https://github.com/user-attachments/assets/1e71a011-4db3-433f-83e6-36d6b76a3d99)

Console font size 16 is okay, the problem here is that font size 16 is different in the command palette.

![console16](https://github.com/user-attachments/assets/b82f5bcf-76b7-49fe-9391-2a4937f54668)

dyphire commented:
> Using a 4K display with 300 % zoom, font size 16 with normal window size is too small, many people have poor eyes, me too.

I can't reproduce this on a 4K monitor, for me the new behavior always keeps the same window footprint. By the way 300 % zoom has no effect on scripts as it is based on window resolution scaling and not HIDPI.

> When the window is maximized, the font size is okay, the problem is normal and maximized window have different font sizes, Before it was the same.

 The old behavior of the extended-menu.lua script where you think the font size is always the same is actually very problematic, so let me give you some examples.

font_size=16 with fullscreen on my test display: too small
<img width="1279" alt="image" src="https://github.com/user-attachments/assets/ea3ff445-8628-4cac-9370-3e5def2600af">
font_size=30 with fullscreen on my test display: small
<img width="1280" alt="image" src="https://github.com/user-attachments/assets/e73b54aa-c9f6-4adb-ac8a-1c13047a4e72">
font_size=56 with fullscreen on my test display: fit
<img width="1280" alt="image" src="https://github.com/user-attachments/assets/6ebd49a6-7518-4bae-805f-6ef94cd6e255">
font_size=56 with default window size on my test display: too big
<img width="589" alt="image" src="https://github.com/user-attachments/assets/c364e2bc-b142-4d34-a9a3-a22c9cef5bb3">
<img width="584" alt="image" src="https://github.com/user-attachments/assets/403a5507-4aaf-468a-b115-99c7e4f898a9">

I think now you can understand why I didn't like it, so I changed its scaling behavior to scale with the window like other 3rd party menus.

> Console font size 16 is okay, the problem here is that font size 16 is different in the command palette.

I can understand that you like the scaling behavior of console.lua, which looks more readable. But console.lua adaptively adjusts the amount of content to be displayed based on the window size, and it does a text measurement to achieve this. The extended-menu.lua doesn't do that, it displays a fixed amount of content across all window sizes, which makes usability much less.

The extended-menu.lua needs to be refactored to behave the same as console.lua, I can't do that, maybe you're interested. When the refactoring is complete, both the old scaling behavior and the current scaling behavior will be superseded.

stax76 commented:
> This is essentially a problem in [extended-menu.lua](https://github.com/Seme4eg/mpv-scripts/blob/98cc63757654e7ede2533af2b0921afbc4165dc1/script-modules/extended-menu.lua#L177-L193) script, which uses dynamic PlayResX and PlayResY values, which causes the font size to have to be dynamic as well or the menu won't be rendered correctly.

But rendering or font problems were never seen by me or reported to me, if I recall correctly. In what way is not rendered correctly?

With the old extended-menu.lua I was getting exactly the same font size in normal, maximized and fullscreen window size, you write that you weren't getting that and even posted screenshots illustrating it. Your font looks not like a Windows default like Segeo UI, maybe that is what is causing the problem.

You prefer or need a modified extended-menu.lua, I prefer the unmodified extended-menu.lua, why not make and option to support both?

I looked at uosc, it's not scaling with the window size, which prefer. It's default font size is perfect, it's not having/using a font size option but instead use font_scale.

dyphire commented:
> You prefer or need a modified extended-menu.lua, I prefer the unmodified extended-menu.lua, why not make and option to support both?

Implement it in the upstream PR and wait for it to merge.

dyphire commented:
> 
> > You prefer or need a modified extended-menu.lua, I prefer the unmodified extended-menu.lua, why not make and option to support both?
> 
> Implement it in the upstream [PR](https://github.com/Seme4eg/mpv-scripts/pull/11) and wait for it to merge.

Done.

stax76 commented:
Thanks, with that, everybody should be able to find a configuration that works.
